test/Constraints/generics.swift

```diff
@@ -1031,3 +1031,27 @@ func test_requirement_failures_in_ambiguous_context() {
 
   f3(A()) // expected-error {{no exact matches in call to local function 'f3'}}
 }
+
+// rdar://106054263 - failed to produce a diagnostic upon generic argument mismatch
+func test_mismatches_with_dependent_member_generic_arguments() {
+  struct Binding<T, U> {}
+  // expected-note@-1 {{arguments to generic parameter 'T' ('Double?' and 'Data.SomeAssociated') are expected to be equal}}
```

Comment:
Cool! Is it not possible to simplify the dependent member types in these cases?

Reply:
Diagnostic engine does (aka) but not very consistently…

Reply:
Yeah I've noticed the same thing. Well, we can look into that separately!

Reply:
Sounds good to me!